src/translator/service.h

```diff
@@ -76,24 +78,28 @@ class Service {
   ///
   /// This is equivalent to a call to:
   /// ```cpp
-  ///    Service(options, AlignedMemory(),  AlignedMemory())
+  ///    Service(options, AlignedMemory(), AlignedMemory(), {})
   /// ```
   /// wherein empty memory is passed and internal flow defaults to file-based
-  /// model, shortlist loading.
+  /// model, shortlist loading. AlignedMemory() corresponds to empty memory
   explicit Service(Ptr<Options> options)
```

Comment:
This is probably not your fault, but any reason why we have 3 different constructors as opposed to using default arguments?

Reply:
I guess the reason for the 3 constructors here is to keep a purely file-based one and a bytearray-based one separated. We can get rid of different constructors in a later PR (addressing https://github.com/browsermt/bergamot-translator/pull/122#discussion_r625868489)

Reply:
There are 2 classes of constructors right now:

1. One that uses `std::string` to pass config ([constructor 1](https://github.com/browsermt/bergamot-translator/blob/main/src/translator/service.h#L92-L94))
    1. This is the constructor that browser extension currently uses (as it is independent of marian internal structs)
2. Other class that uses `marian::Options` to pass config ([constructor 2](https://github.com/browsermt/bergamot-translator/blob/main/src/translator/service.h#L83-L84) and [constructor 3](https://github.com/browsermt/bergamot-translator/blob/main/src/translator/service.h#L69-L70)). I believe @jerinphilip can throw some light on why we need these ones 👍 
    1. I think we can merge [constructor 2](https://github.com/browsermt/bergamot-translator/blob/main/src/translator/service.h#L83-L84) and [constructor 3](https://github.com/browsermt/bergamot-translator/blob/main/src/translator/service.h#L69-L70) into one
    2. Probably, we should make this constructor private because `Service` class now is the main API class and we wouldn't want to expose marian internal structs in public APIs. However, I am fine with keeping it public if there is a reason for it.

Reply:
> This is probably not your fault

This is my fault, and part of me still thinks the fault should stay. NULL values (empty `AlignedMemory`) as default argument is what should go away (this is kludge added to get green tick on CI not breaking a late `TranslationModel`), with `Service` accepting only legit bytearray payloads thus making way for two constructors (one from options with file-loads and one from options + bytearray payload). I'm not going to count the delegated constructor, it has only a minimal maintenance overhead. 

> We can get rid of different constructors in a later PR

Mark this as resolved?